### PR TITLE
Gml port

### PR DESCRIPTION
**Pull request by Matyrobbrt**
GML port, `mods.groovy` and move to test source set.

### CODE REVIEW

File: src/test/groovy/groovylicioustest/Configs.groovy [hunk 1]
```diff
@@ -9,19 +9,25 @@ import net.minecraftforge.common.ForgeConfigSpec
 @CompileStatic
 class Configs {
 
-    //@Config
+    @Config
+    @CompileStatic
```
Comment (PaintNinja) [inline src/test/groovy/groovylicioustest/Configs.groovy]: As far as I'm aware, CompileStatic applies to all sub-classes, so you shouldn't need to individually annotate each sub-class with CompileStatic here.

File: src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy [hunk 2]
```diff
@@ -60,7 +60,7 @@ class ConfigASTTransformation extends AbstractASTTransformation {
 
         // get the configType and modId from the annotation
         configType = getMemberConfigType(configAnnotation, configDataClass)
-        modId = getMemberStringValue(configAnnotation, 'modId', '$unknown')
+        modId = getMemberStringValue(configAnnotation, 'modId', ModRegistry.getData(configDataClass.getPackageName())?.modId())
```
Comment (PaintNinja) [inline src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy:63]: The `if (modId == '$unknown') {` on ln 152 of this file needs removing also

File: src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy [hunk 3]
```diff
@@ -101,7 +101,7 @@ class ConfigASTTransformation extends AbstractASTTransformation {
                 // Looks like the @Mojo/@Mod is in a different file, let's register a transform to the
                 // GroovyliciousMojoTransformRegistry to add a static { configDataClass.init() } to the Mod's main class
                 if (DEBUG) println "Adding transform to @GroovyliciousMod"
-                GroovyliciousMojoTransformRegistry.addTransform { AnnotationNode modAnnotation, ClassNode modClass ->
+                GModASTTransformer.registerTransformer { ClassNode modClass, AnnotationNode modAnnotation, SourceUnit source$ ->
```
Comment (PaintNinja) [inline src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy:104]: Is there still a need for the GroovyliciousMojoTransformRegistry or does GModASTTransformer.registerTransformer replace it?

File: src/main/groovy/ga/ozli/minecraftmods/groovylicious/Groovylicious.groovy [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
-import net.thesilkminer.mc.austin.api.Mod
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
+import groovy.util.logging.Slf4j
 
 @POJO
+@GMod(MOD_ID)
 @CompileStatic
-@Mod(MOD_ID)
+@Slf4j(category = MOD_ID)
```
Comment (PaintNinja) [inline src/main/groovy/ga/ozli/minecraftmods/groovylicious/Groovylicious.groovy]: image
I didn't do this as it partially breaks logging in prod

Comment (Matyrobbrt) [reply src/main/groovy/ga/ozli/minecraftmods/groovylicious/Groovylicious.groovy]: That... doesn't seem right

File: src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy (hunk 3 above)
Comment (Matyrobbrt) [reply src/transform/groovy/ga/ozli/minecraftmods/groovylicious/transform/config/ConfigASTTransformation.groovy:104]: `GModASTTransformer` / `GModTransformer` effectively replaces it, yes.

**Pull request merged**